crates/astria-sequencer/src/app.rs

```diff
@@ -362,21 +373,40 @@ impl App {
     /// Since transaction execution now happens in the proposal phase, results
     /// are cached in the app and returned here during the usual ABCI block execution process.
     ///
+    /// If the tx was not executed during the proposal phase it will be executed here.
+    ///
     /// Note that the first two "transactions" in the block, which are the proposer-generated
     /// commitments, are ignored.
-    #[instrument(name = "App::deliver_tx_after_execution", skip_all, fields(
-        tx_hash =  %telemetry::display::hex(tx_hash),
+    #[instrument(name = "App::deliver_tx_after_proposal", skip_all, fields(
+        tx_hash =  %telemetry::display::hex(&Sha256::digest(&tx.tx)),
     ))]
-    pub(crate) fn deliver_tx_after_execution(
+    pub(crate) async fn deliver_tx_after_proposal(
         &mut self,
-        tx_hash: &[u8; 32],
+        tx: abci::request::DeliverTx,
     ) -> Option<anyhow::Result<Vec<abci::Event>>> {
         if self.processed_txs < 2 {
             self.processed_txs += 1;
             return Some(Ok(vec![]));
         }
 
-        self.execution_result.remove(tx_hash)
+        // When the hash is not empty, we have already executed and cached the results
+        if !self.executed_proposal_hash.is_empty() {
+            let tx_hash: [u8; 32] = sha2::Sha256::digest(&tx.tx).into();
+            return self.execution_result.remove(&tx_hash);
+        }
+
+        let signed_tx = match signed_transaction_from_bytes(&tx.tx) {
+            Err(e) => {
+                debug!(
+                    error = AsRef::<dyn std::error::Error>::as_ref(&e),
+                    "failed to decode deliver tx payload to signed transaction; ignoring it",
+                );
+                return None;
+            }
+            Ok(tx) => tx,
+        };
```

Comment:
```suggestion
        let signed_tx = signed_transaction_from_bytes(&tx.tx).map_err(|e|
                debug!(
                    error = AsRef::<dyn std::error::Error>::as_ref(&e),
                    "failed to decode deliver tx payload to signed transaction; ignoring it",
                );
                None
            )?;
```

debating whether this should error or not - we should never get an un-deserializable tx

Reply:
i guess returning `None` hits the `expect` in the caller, so that's fine

Reply:
I don't think this behavior should be changed in this PR, maintaining the previous behavior

Reply:
Yeah this maintains parity with what happens in the return above if there is an issue with execution.